src/app/frontend/common/services/create/service.ts

```diff
@@ -96,7 +96,9 @@ export class CreateService {
       this.reportError(i18n.MSG_DEPLOY_DIALOG_ERROR, error.error);
       throw error;
     } else {
-      this.router_.navigate(['overview']);
+      this.router_.navigate(['overview'], {
+        queryParams: {[NAMESPACE_STATE_PARAM]: spec.namespace},
```

Comment:
This approach is followed in deploy() method which is used in create from form

https://github.com/kubernetes/dashboard/blob/7680a7915a744265051be8e082004a0c61f6ccb1/src/app/frontend/common/services/create/service.ts#L128

Reply:
@floreks  , the namespace assigned in the spec for CreateContent is anyways the current namespace. I didn't realize to point it out earlier.

This is also because in create from input, there isn't an option to explicitly mention the ns. It's current by default

https://github.com/kubernetes/dashboard/blob/7680a7915a744265051be8e082004a0c61f6ccb1/src/app/frontend/common/services/create/service.ts#L71

Reply:
@anshulahuja98 In you choose create from input/file then it's very easy to override the namespace via the resource definition.

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    k8s-app: kubernetes-dashboard
  name: kubernetes-dashboard
  namespace: xyz
```
Then the namespace from the file will be taken to create a new resource. That's why changing the namespace to the namespace of the resource might not be a good idea. It's sometimes hidden from the user eyes and possibly not really expected.

Reply:
Note that when trying to create a resource specifying another namespace from the actual namespace, you will get an error "Deploying file has failed: MSG_DEPLOY_NAMESPACE_MISMATCH_ERROR"

So in case of success, the current namespace and the namespace of the created resource will be the same one

Reply:
Yes, I think it might be more appropriate to keep the current namespace. If the user fails to create resource from yaml, switching to the namespace in yaml will cause bad results. It's also not good to get different solutions in different kind of cases, which will make users confused.